src/kernels/jupyter/remoteKernelFinder.ts

```diff
@@ -139,8 +143,11 @@ export class RemoteKernelFinder implements IRemoteKernelFinder {
     }
 
     private async getInterpreter(spec: IJupyterKernelSpec, baseUrl: string) {
-        const parsed = new url.URL(baseUrl);
-        if (parsed.hostname.toLocaleLowerCase() === 'localhost' || parsed.hostname === '127.0.0.1') {
+        const parsed = new URL(baseUrl);
+        if (
+            (parsed.hostname.toLocaleLowerCase() === 'localhost' || parsed.hostname === '127.0.0.1') &&
```

Comment:
Do we need something more general here? Like so for edge cases? https://www.npmjs.com/package/is-localhost-ip

Reply:
Oh probably. Separate issue though. This is just preventing it using the interpreter service when there's no python extension.

Reply:
Created a new issue #9748